datastream-kafka-connector/src/main/java/com/linkedin/datastream/connectors/kafka/AbstractKafkaBasedConnectorTask.java

```diff
@@ -165,9 +161,30 @@ protected AbstractKafkaBasedConnectorTask(KafkaBasedConnectorConfig config, Data
     _pausePartitionOnError = config.getPausePartitionOnError();
     _pauseErrorPartitionDuration = config.getPauseErrorPartitionDuration();
     _enableAdditionalMetrics = config.getEnableAdditionalMetrics();
-    _startOffsets = Optional.ofNullable(_datastream.getMetadata().get(DatastreamMetadataConstants.START_POSITION))
-        .map(json -> JsonUtils.fromJson(json, new TypeReference<Map<Integer, Long>>() {
-        }));
+
+    _startOffsets = new HashMap<>();
+    String json = _datastream.getMetadata().get(DatastreamMetadataConstants.START_POSITION);
+    if (StringUtils.isNotBlank(json)) {
+      _startOffsets.putAll(JsonUtils.fromJson(json, new TypeReference<Map<Integer, Long>>() {
+      }));
+    }
+
+    // if this datastream wants to use specific start offsets, we need to have the auto.offset.reset config
+    // set to "none" in the kafka consumer configs, so that the Kafka consumer throws NoOffsetForPartitionException
+    // upon first poll, to be handled by seeking to teh startOffsets
+    if (!_startOffsets.isEmpty()) {
+      _logger.info("Datastream contains startOffsets, setting {} = \"{}\" in consumer configs "
```

Comment:
In case someone adds the datastream metadata too, should we print that here too? It'll be helpful in debugging if someone sets this to something else and is wondering why it isn't taking effect.

`String strategy = _datastream.getMetadata().get(KafkaDatastreamMetadataConstants.CONSUMER_OFFSET_RESET_STRATEGY);`

This log is only printing the config.